### PR TITLE
Coding standard with PHP 8 features

### PR DESCRIPTION
**Pull request by kukulich**
Each sniff enabled in separate commit so it's easy to remove the sniff if not approved.

### CODE REVIEW

**Comment by kukulich** [comment]
@WyriHaximus For you info.

**Comment by kukulich** [comment]
@Ocramius Next one :)

File: src/NodeCompiler/CompileNodeToValue.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ class CompileNodeToValue
      *
      * @throws Exception\UnableToCompileNode
      */
-    public function __invoke(Node $node, CompilerContext $context)
+    public function __invoke(Node $node, CompilerContext $context): string|int|float|bool|array|null
```
Comment (Ocramius) [inline src/NodeCompiler/CompileNodeToValue.php:32]: Ooooh, I like how `scalar` was implemented here :-)

File: test/unit/Reflection/ReflectionFunctionTest.php [hunk 2]
```diff
@@ -119,18 +119,22 @@ public function testStaticCreationFromName(): void
 
     public function testCreateFromClosure(): void
     {
-        $myClosure  = static function () {
+        // phpcs:disable SlevomatCodingStandard.Functions.RequireArrowFunction
```
Comment (Ocramius) [inline test/unit/Reflection/ReflectionFunctionTest.php:122]: Good catch not to change these 👍

File: test/unit/Reflection/ReflectionTypeTest.php [hunk 3]
```diff
@@ -18,8 +18,7 @@
  */
 class ReflectionTypeTest extends TestCase
 {
-    /** @var Reflector|MockObject */
```
Comment (Ocramius) [inline test/unit/Reflection/ReflectionTypeTest.php:21]: Not important, but we'll need to use `&` here in future.

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Awesome, thanks! 👍

**Pull request merged**